e2e/rbd.go

```diff
@@ -487,6 +489,96 @@ var _ = Describe("RBD", func() {
 				// validate created backend rbd images
 				validateRBDImageCount(f, 0, defaultRBDPool)
 			})
+			By("create a Block mode RWOP PVC and bind it to more than one app", func() {
+				if k8sVersionGreaterEquals(f.ClientSet, 1, 22) {
+					pvc, err := loadPVC(rawPVCRWOPPath)
+					if err != nil {
+						e2elog.Failf("failed to load PVC: %v", err)
+					}
+					pvc.Namespace = f.UniqueName
+
+					app, err := loadApp(rawAppPath)
+					if err != nil {
+						e2elog.Failf("failed to load application: %v", err)
+					}
+					app.Namespace = f.UniqueName
+					aname := app.Name
+					err = createPVCAndvalidatePV(f.ClientSet, pvc, deployTimeout)
+					if err != nil {
+						e2elog.Failf("failed to create PVC: %v", err)
+					}
+					// validate created backend rbd images
+					validateRBDImageCount(f, 1, defaultRBDPool)
+
+					err = createApp(f.ClientSet, app, deployTimeout)
+					if err != nil {
+						e2elog.Failf("failed to create application: %v", err)
+					}
+					// create one more  app with same PVC
+					name := fmt.Sprintf("%s%d", f.UniqueName, deployTimeout)
+					app.Name = name
+					err = createAppErr(f.ClientSet, app, deployTimeout, ErrRWOPConflict)
+					if err != nil {
+						e2elog.Failf("application should not go to running state due to RWOP access mode")
+					}
+					err = deletePod(name, app.Namespace, f.ClientSet, deployTimeout)
+					if err != nil {
+						e2elog.Failf("failed to delete application: %v", err)
```

Comment:
the logic to validate RWOP remains the same for both rbd and cephfs can we have a common function for both to avoid code duplication?

Reply:
done.